src/state_management.cpp

```diff
@@ -7,6 +7,12 @@ namespace ara::sm {
     void StateManagement::Work() {
         std::cout << "Starting work" << std::endl;
         while (!killFlag) {
+            if (stateClient != nullptr) {
```

Comment:
This should be one line higher -  before while I suppose? and why are we even calling this function?
It should be set by EM - in our version, we can set it in test class constructor.
Look at figure 7.1 in SWS_StateManagement.
I think here we should call GetInitialMachineStateTransitionResult and if it returns TransitionFailed, we can call MachineSetState(Restart). Otherwise, we can set our Function Groups to the "On" state.

Reply:
Done